netmaster/k8snetwork/networkpolicy.go

```diff
@@ -135,204 +257,489 @@ func (k8sNet *k8sContext) deleteEpg(networkname, epgName, policyName string) err
 
-func (k8sNet *k8sContext) createRule(policyName, ruleID, action string) error {
-	npLog.Infof("create rule %s[%s] [%s]", policyName, ruleID, action)
+//Post rule  to contiv if not exist
+func (k8sNet *k8sContext) createRule(cRule *client.Rule) error {
 
-	if val, err := k8sNet.contivClient.RuleGet(defaultTenantName, policyName, ruleID); err == nil {
-		if val.Action != action {
-			k8sNet.deleteRule(policyName, ruleID)
+	if val, err := k8sNet.contivClient.RuleGet(cRule.TenantName,
+		cRule.PolicyName, cRule.RuleID); err == nil {
+		if val.Action != cRule.Action {
+			k8sNet.deleteRule(cRule.TenantName,
+				cRule.PolicyName, cRule.RuleID)
 		} else {
+			npLog.Infof("Rule:%+v already exist", *cRule)
 			return nil
 		}
 	}
 
-	if err := k8sNet.contivClient.RulePost(&client.Rule{
-		TenantName: defaultTenantName,
-		PolicyName: policyName,
-		RuleID:     ruleID,
-		Direction:  "in",
-		Action:     action,
-	}); err != nil {
-		npLog.Errorf("failed to create rule-id [%s] %s", ruleID, err)
+	if err := k8sNet.contivClient.RulePost(cRule); err != nil {
+		npLog.Errorf("failed to create rule: %s, %v", cRule.RuleID, err)
 		return err
 	}
 
 	for func() error {
-		_, err := k8sNet.contivClient.RuleGet(defaultTenantName, policyName, ruleID)
+		_, err := k8sNet.contivClient.RuleGet(cRule.TenantName,
+			cRule.PolicyName, cRule.RuleID)
 		return err
 	}() != nil {
 		time.Sleep(time.Millisecond * 100)
 	}
 	return nil
 }
 
-func (k8sNet *k8sContext) deleteRule(policyName, ruleID string) error {
-	npLog.Infof("delete rule-id %s", ruleID)
+//Delete rule from contiv system
+func (k8sNet *k8sContext) deleteRule(tenantName string,
+	policyName, ruleID string) error {
+	npLog.Infof("Delete rule: %s:%s", ruleID, policyName)
 
-	if _, err := k8sNet.contivClient.RuleGet(defaultTenantName, policyName, ruleID); err != nil {
+	if _, err := k8sNet.contivClient.
+		RuleGet(tenantName, policyName, ruleID); err != nil {
 		return nil
 	}
 
-	if err := k8sNet.contivClient.RuleDelete(defaultTenantName, policyName, ruleID); err != nil {
-		npLog.Errorf("failed to delete rule %s[%s], %s", policyName, ruleID, err)
+	if err := k8sNet.contivClient.
+		RuleDelete(tenantName, policyName, ruleID); err != nil {
+		npLog.Errorf("Failure rule del Ops:%s:%s,%v",
+			ruleID, policyName, err)
 		return err
 	}
 
 	for func() error {
-		_, err := k8sNet.contivClient.RuleGet(defaultTenantName, policyName, ruleID)
+		_, err := k8sNet.contivClient.
+			RuleGet(tenantName, policyName, ruleID)
 		return err
 	}() == nil {
 		time.Sleep(time.Millisecond * 100)
 	}
 	return nil
 }
 
-func (k8sNet *k8sContext) getIsolationPolicy(annotations map[string]string) string {
-	var inPolicy struct {
-		Ingress map[string]string `json:"ingress"`
+//Sub handler to process  Network Policy event from K8s srv
+func (k8sNet *k8sContext) processK8sNetworkPolicy(
+	opCode watch.EventType, np *network_v1.NetworkPolicy) {
+	if np.Namespace == "kube-system" { //not applicable for system namespace
+		return
 	}
```

Comment:
minor, but since both this and processK8sPod both do this check, you could do it in processK8sEvent instead?